source/common/network/connection_impl.h

```diff
@@ -74,7 +74,7 @@ class ConnectionImpl : public virtual Connection,
   uint32_t readBufferLimit() const override { return read_buffer_limit_; }
 
   // Network::BufferSource
-  Buffer::Instance& getReadBuffer() override { return read_buffer_; }
+  Buffer::Instance& getReadBuffer() override { return *read_buffer_; }
   Buffer::Instance& getWriteBuffer() override { return *current_write_buffer_; }
```

Comment:
Yeah, that whole situation is not great but per @alyssawilk the comment should have the details.